daffodil-runtime1-layers/src/main/scala/org/apache/daffodil/layers/runtime1/LineFoldedTransformer.scala

```diff
@@ -393,11 +298,25 @@ class LineFoldedOutputStream(mode: LineFoldMode, jos: OutputStream) extends Outp
     }
   }
 
+  // TODO: This is broken because an isolated \r or \n is output as \r\n, but without verifying that the next character
+  //  is space or tab. If that ends up as a \r\n not followed by a space or tab it has broken the data, as on a reparse
+  //  that \r\n will be interpreted as an end-of-layer marker.
```

Comment:
That's true only if the \r\n is being used as a boundaryMark. If the layerLengthKind is 'implicit' then this insertion of \r\n doesn't harm anything. 

Maybe line folded layers don't need boundaryMark support at all? Maybe implicit is enough?